`Checkbox` - Updating documentation for indeterminate state (#1726)
hashicorp/design-system: website/docs/components/form/checkbox/partials/guidelines/guidelines.md

- When only one choice can be selected, consider [Radio](/components/form/radio/) buttons.
- When checking or unchecking results in an immediate change, consider [Toggle](/components/form/toggle).

> ### Indeterminate State

Reviewer: @andgen404 Can you please also update the headings for "Layout" and "Required and optional" to H2s as well so they aren't nested under the "Indeterminate state" section?

> Checkboxes, primarily toggling between checked and unchecked states, offer a third, indeterminate state for indicating a "partially checked" condition in nested structures, manageable not through HTML, but JavaScript.

Reviewer: This feels a little verbose and technical for the Design Guidelines. Instead, we should lean into providing proper usage guidance for the specific topic. I also suggest adding a few visual examples of proper and improper usage. 

```suggestion
Indeterminate states indicate a "partially checked" condition in nested structures or tables. 
```

Reviewer 2: '_manageable not through HTML, but JavaScript._' unsure of the value of this statement in the guidelines section – I'm tempted to leave it out.

Reviewer 3: @alex-ju also, should we mention the fact that the `indeterminate={{true}}` is an Ember thing (and potentially link to some documentation for it) and not a native HTML feature?

Reviewer 2: If we feel it's important to highlight that this is managed by the component via Ember I can add it [to the code section](https://helios.hashicorp.design/components/form/checkbox?tab=code#indeterminate).

Reviewer 3: Probably yes: as a developer (expecially coming from other frameworks) I would struggle to understand how this indeterminate works (also because is missing the `@` typical of Ember arguments.

Reviewer 2: cool, I'll add a commit to this branch (if that's ok with you, Andrew)

Author: Works for me 🙂

Reviewer 2: @didoo added an extra paragraph in 2325b2e; we also touched on this in the API section

- When only one choice can be selected, consider [Radio](/components/form/radio/) buttons.
- When checking or unchecking results in an immediate change, consider [Toggle](/components/form/toggle).

## Indeterminate state

Indeterminate states indicate a "partially checked" condition in nested structures or tables. 

!!! Do

![Example of proper usage of the indeterminate state in the checkbox] (/assets/components/form/checkbox/checkbox-indeterminate-do.png)
!!!

!!! Dont

> ![Example of misuse of the indeterminate state in the checkbox] (/assets/components/form/checkbox/checkbox-indeterminate-dont.png)

Author: ```suggestion
![Example of misuse of the indeterminate state in the checkbox] (/assets/components/form/checkbox/checkbox-Indeterminate-dont.png)
```

- When only one choice can be selected, consider [Radio](/components/form/radio/) buttons.
- When checking or unchecking results in an immediate change, consider [Toggle](/components/form/toggle).

## Indeterminate state

Indeterminate states indicate a "partially checked" condition in nested structures or tables. 

> !!! Do

Reviewer 3: Love the Do/Don't illustrations, they give clear guidance. 👌

Reviewer 4: [Suggestion] One other example that it might be useful to include (which is a common one) is in the context of a form. [Could be something like this](https://www.figma.com/file/42LK10XbP5IERhzzgMOiI2/Website-assets?type=design&node-id=2545%3A2339&mode=design&t=yLvHHraGqVWkZRDl-1)
